File: backend/app/schemas/storage.py
"""Storage schemas."""

from pydantic import BaseModel
from typing import List


class LocalStorageResponse(BaseModel):
    """Local storage information."""
    disk_total_bytes: int
    disk_used_bytes: int
    disk_free_bytes: int
    disk_used_percent: float
    episodes_bytes: int
    converted_bytes: int
    total_podcast_bytes: int
    episodes_mb: float
    converted_mb: float
    total_podcast_mb: float


class PodcastStorageItem(BaseModel):
    """Storage info for a single podcast."""
    podcast_id: int
    podcast_title: str
    total_bytes: int
    total_mb: float
    episode_count: int
    synced_count: int


class StorageByPodcastResponse(BaseModel):
    """Storage breakdown by podcast."""
    podcasts: List[PodcastStorageItem]


class CleanupRequest(BaseModel):
    """Cleanup request parameters."""
    cleanup_type: str  # "age", "storage_limit", "failed", "orphaned"
    days_old: int = 30
    max_storage_mb: int = 1000
    keep_synced: bool = True


class CleanupResponse(BaseModel):
    """Cleanup operation result."""
    success: bool
    items_deleted: int
    bytes_freed: int
    mb_freed: float
    message: str

File: frontend/src/api/storage.ts
import { apiClient } from './client'

export interface LocalStorageInfo {
  disk_total_bytes: number
  disk_used_bytes: number
  disk_free_bytes: number
  disk_used_percent: number
  episodes_bytes: number
  converted_bytes: number
  total_podcast_bytes: number
  episodes_mb: number
  converted_mb: number
  total_podcast_mb: number
}

export interface PodcastStorageItem {
  podcast_id: number
  podcast_title: string
  total_bytes: number
  total_mb: number
  episode_count: number
  synced_count: number
}

export interface StorageByPodcast {
  podcasts: PodcastStorageItem[]
}

export interface CleanupRequest {
  cleanup_type: 'age' | 'storage_limit' | 'failed' | 'orphaned'
  days_old?: number
  max_storage_mb?: number
  keep_synced?: boolean
}

export interface CleanupResult {
  success: boolean
  items_deleted: number
  bytes_freed: number
  mb_freed: number
  message: string
}

export const storageApi = {
  // Get local storage info
  getLocalStorage: async (): Promise<LocalStorageInfo> => {
    const response = await apiClient.get<LocalStorageInfo>('/api/storage/local')
    return response.data
  },

  // Get storage by podcast
  getStorageByPodcast: async (): Promise<StorageByPodcast> => {
    const response = await apiClient.get<StorageByPodcast>('/api/storage/by-podcast')
    return response.data
  },

  // Cleanup storage
  cleanup: async (request: CleanupRequest): Promise<CleanupResult> => {
    const response = await apiClient.post<CleanupResult>('/api/storage/cleanup', request)
    return response.data
  },
}
